src/components/KeyboardShortcuts/index.spec.tsx

```diff
@@ -0,0 +1,181 @@
+      <KeyboardShortcuts {...props} />,
+      KeyboardShortcutsBase,
+      {
+        shallowOptions: { context: { store } },
+      },
+    );
+  };
+
+  it('renders a list of keyboard shortcuts', () => {
+    const root = render();
+
+    expect(root.find(`.${styles.KeyboardShortcuts}`)).toHaveLength(1);
+  });
+
+  it('binds and unbinds keydown to the listener', () => {
+    document.addEventListener = jest.fn();
+    document.removeEventListener = jest.fn();
+
+    const root = render();
+    const { keydownListener } = root.instance() as KeyboardShortcutsBase;
+
+    expect(document.addEventListener).toHaveBeenCalledWith(
+      'keydown',
+      keydownListener,
+    );
+
+    root.unmount();
+
+    expect(document.removeEventListener).toHaveBeenCalledWith(
+      'keydown',
+      keydownListener,
+    );
+  });
+
+  it.each([
+    ['previous', 'k', RelativePathPosition.previous],
+    ['next', 'j', RelativePathPosition.next],
+  ])(
+    'dispatches goToRelativeFile with %s when "%s" is pressed',
+    (direction, key, position) => {
+      const currentPath = 'file1.js';
+      const pathList = [currentPath];
+      const versionId = 123;
+
+      const store = configureStore();
+      const dispatch = spyOn(store, 'dispatch');
+      const fakeThunk = createFakeThunk();
+      const _goToRelativeFile = fakeThunk.createThunk;
+
+      const root = render({
+        _goToRelativeFile,
+        currentPath,
+        pathList,
+        store,
+        versionId,
+      });
+
+      const { keydownListener } = root.instance() as KeyboardShortcutsBase;
+
+      // TS cannot seem to figure out that `key` is a string.
+      // @ts-ignore
+      const event = createFakeKeyboardEvent({ key }) as KeyboardEvent;
+
+      keydownListener(event);
+
+      expect(dispatch).toHaveBeenCalledWith(fakeThunk.thunk);
+      expect(_goToRelativeFile).toHaveBeenCalledWith({
+        currentPath,
+        pathList,
+        position,
+        versionId,
+      });
+    },
+  );
+
+  it('dispatches expandTree when "e" is pressed', () => {
+    const versionId = 123;
+
+    const store = configureStore();
+    const dispatch = spyOn(store, 'dispatch');
+
+    const root = render({
+      store,
+      versionId,
+    });
+
+    const { keydownListener } = root.instance() as KeyboardShortcutsBase;
+
+    const event = createFakeKeyboardEvent({ key: 'e' }) as KeyboardEvent;
+
+    keydownListener(event);
+
+    expect(dispatch).toHaveBeenCalledWith(
+      versionActions.expandTree({ versionId }),
+    );
+  });
+
+  it('dispatches collapseTree when "c" is pressed', () => {
+    const versionId = 123;
+
+    const store = configureStore();
+    const dispatch = spyOn(store, 'dispatch');
+
+    const root = render({
+      store,
+      versionId,
+    });
+
+    const { keydownListener } = root.instance() as KeyboardShortcutsBase;
+
+    const event = createFakeKeyboardEvent({ key: 'c' }) as KeyboardEvent;
+
+    keydownListener(event);
```

Comment:
How about `renderAndTriggerKeyEvent(props, event)`? I think that would cut down on duplication across a couple tests.

Reply:
Good idea!